File: 01_insertBegDLL/1.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
        self.prev = None

def create_node(data):
    return Node(data)

def insert_at_front(head, data):
    new_node = create_node(data)
    new_node.next = head
    if head is not None:
        head.prev = new_node
    return new_node

def print_list(head):
    temp = head
    while temp:
        print(temp.data, end=' ')
        temp = temp.next
    print()

head = create_node(2)
second = create_node(3)
third = create_node(4)

head.next = second
second.prev = head
second.next = third
third.prev = second

print("Original Linked List:")
print_list(head)

print("After inserting Node at the front:")
head = insert_at_front(head, 1)
print_list(head)

File: 01_insertBegDLL/1.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node *next;
    struct Node *prev;
};

struct Node *createNode(int new_data) {
    struct Node *newnode = (struct Node *)malloc(sizeof(struct Node));
    newnode->data = new_data;
    newnode->next = NULL;
    newnode->prev = NULL;
    return newnode;
}

struct Node *insertAtFront(struct Node *head, int new_data) {

    struct Node *newnode = createNode(new_data);

    newnode->next = head;

    if (head != NULL) {
        head->prev = newnode;
    }

    return newnode;
}


void printList(struct Node *head) {
    struct Node *temp = head;
    while (temp != NULL) {
        printf(" %d", temp->data);
        temp = temp->next;
    }
    printf("\n");
}

int main() {


    struct Node *head = createNode(2);
    head->next = createNode(3);
    head->next->prev = head;
    head->next->next = createNode(4);
    head->next->next->prev = head->next;

    printf("Original Linked List:");
    printList(head);

    printf("After inserting Node at the front:");
    int data = 1;
    head = insertAtFront(head, data);

    printList(head);

    return 0;
}